File: course/src/20_network_and_sockets/io_streams/copybytes.py
# Python equivalent of the Java CopyBytes program

def copy_bytes():
    # Open the input and output files
    try:
        with open("xanadu.txt", "rb") as in_file, open("outagain.txt", "wb") as out_file:
            # Read and write byte-by-byte
            while (byte := in_file.read(1)):
                out_file.write(byte)
    except IOError as e:
        print(f"Error: {e}")

# Call the function to copy bytes
copy_bytes()

File: course/src/04_concurrency/20_network_and_sockets/io_streams/CopyBytes.java
// input stream
import java.io.FileInputStream;
// output stream
import java.io.FileOutputStream;
// io exception
import java.io.IOException;

public class CopyBytes {
    public static void main(String[] args) throws IOException {
        
        // starts at null and after receive something?
        FileInputStream in = null;
        FileOutputStream out = null;

        try {
            in = new FileInputStream("xanadu.txt");
            out = new FileOutputStream("outagain.txt");
            int c;

            while ((c = in.read()) != -1) {
                out.write(c);
            }
        } finally {
            // close the streams
            if (in != null) {
                in.close();
            }
            if (out != null) {
                out.close();
            }
        }
    }
}

